### PR TITLE
Project tagging breakup

### PR DESCRIPTION
**Pull request by lorenjohnson**
- [x] need to format children of a post, in the case of e.g. /noo/post/43?children=true. 
we currently just send the entire object as seen in the database back

### CODE REVIEW

**Comment by Connoropolous** [comment]
@levity if I could get your eyes on this (the data migration) in the near future, before the rest of the review that'd be great :) 
https://github.com/Hylozoic/hylo-node/pull/209/commits/75fd7fb9d446a27bc3c4268b91acce901e449890

File: lib/migration.js [hunk 1]
```diff
@@ -104,3 +104,59 @@ export const convertAllProjectsToPosts = () => {
...
+    Promise.resolve())
+  )
+}
+
+const findOfficialProjectTags = (projects) => {
+  return reduce(
+    projects,
+    (promise, project) => promise
+      .then(() => PostTag.query(qb => {
+          qb.where('post_id', '=', project.id)
```
Comment (levity) [inline lib/migration.js]: you don't need to specify `=`; if you omit it, it's assumed that's what you mean

File: api/controllers/PostController.js [hunk 2]
```diff
@@ -102,6 +102,13 @@ const queryForFollowed = function (req, res) {
   }))
 }
 
+const queryForPost = function (req, res) {
```
Comment (levity) [inline api/controllers/PostController.js]: Since this is used for a child post I would name this `queryForChildPost`.

File: api/models/Tag.js [hunk 3]
```diff
@@ -279,7 +279,7 @@ module.exports = bookshelf.Model.extend({
   },
 
   nonexistent: (names, communityIds) => {
-    if (names.length === 0 || communityIds.length === 0) return Promise.resolve({})
+    if (names.length === 0 || !communityIds || communityIds.length === 0) return Promise.resolve({})
```
Comment (levity) [inline api/models/Tag.js]: using lodash's `isEmpty` here would cover both the null and empty cases

Comment (Connoropolous) [inline api/models/Tag.js]: k, will switch!

Comment (Connoropolous) [inline api/models/Tag.js]: oh, actually, the two cases are opposite, it needs both separate

Comment (Connoropolous) [inline api/models/Tag.js]: hm, sorry again. I COULD switch it but I think it would make it less clear. So I'm still going to leave it

**Comment by Connoropolous** [comment]
bing! it's ready

**Pull request merged**